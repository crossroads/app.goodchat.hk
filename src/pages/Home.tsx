import {
  IonContent,
  IonHeader,
  IonPage,
  IonTitle,
  IonToolbar,
} from "@ionic/react";
import React from "react";

const Home: React.FC = () => {
  return (
    <IonPage title="home">
      <IonHeader>
        <IonToolbar>
          <IonTitle>Home</IonTitle>
        </IonToolbar>
      </IonHeader>
    </IonPage>
  );
};

export default Home;
