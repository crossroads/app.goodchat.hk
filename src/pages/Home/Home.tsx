import {
  IonButton,
  IonButtons,
  IonHeader,
  IonPage,
  IonTitle,
  IonToolbar,
} from "@ionic/react";
import React from "react";
import useAuth from "hooks/useAuth/useAuth";
import { AUTHENTICATED } from "config/localStorageKeys";

const Home: React.FC = () => {
  const { setIsAuthenticated } = useAuth();

  const handleLogout = () => {
    setIsAuthenticated(false);
    localStorage.removeItem(AUTHENTICATED);
  };

  return (
    <IonPage title="home">
      <IonHeader>
        <IonToolbar>
          <IonTitle>Home</IonTitle>
          <IonButtons slot="end">
            <IonButton onClick={handleLogout}>Log out</IonButton>
          </IonButtons>
        </IonToolbar>
      </IonHeader>
    </IonPage>
  );
};

export default Home;
