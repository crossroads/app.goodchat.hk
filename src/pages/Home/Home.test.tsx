import React from "react";
import { render } from "@testing-library/react";
import Home from "pages/Home/Home";

import {
  expectToRenderHeaderWithTitle,
  expectToRenderLogoutButtonAtHeaderEnd,
} from "test-utils/matchers";

test("renders without crashing", () => {
  const { container } = render(<Home />);
  expect(container).toBeInTheDocument();
});

expectToRenderHeaderWithTitle({
  element: <Home />,
  title: "Home",
});

expectToRenderLogoutButtonAtHeaderEnd(<Home />);
