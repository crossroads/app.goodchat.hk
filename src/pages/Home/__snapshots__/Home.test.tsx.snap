// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders a log out button at the end of the header 1`] = `
<ion-buttons
  slot="end"
>
  <ion-button>
    Log out
  </ion-button>
</ion-buttons>
`;
